code/hof-parameter.kt

```diff
@@ -1,7 +1,8 @@
-fun transform(initial: String, f: (String) -> String) = f(initial)
+fun transform(initial: String, transformer: (String) -> String) = transformer(initial)
 
-val result = transform("hello", { x -> x.toUpperCase() })
+// Function reference
+val result = transform("hello", String::uppercase)
```

Comment:
Perhaps, from the example's perspective, changing the functionality slightly, e.g. to `{ "PREFIX_${it.uppercase()}" },` would make more sense. The point is not to showcase the shortest example, but to demonstrate how functions are used. I can make the corresponding change to the C# example later to match with this modification.